### PR TITLE
feat(auth): real-time field validation on Login and Signup screens

### LINKED ISSUE #321: Validate name, email, and password fields as user types on Login and Signup screens
## Context

Login and Signup screens currently validate fields only on form submission. Users get no feedback until they tap the button.

## Scope

- Validate email, password fields on `LoginScreen` as the user edits them
- Validate email, password, confirm password, and profile name fields on `SignupScreen` as the user edits them
- Reuse existing validator interfaces (`CredentialsValidator`, `ProfileNameValidator`) — do not duplicate rules in the ViewModel
- Display inline field errors below each field; clear the error as soon as the field becomes valid again

## Out of scope

- Changes to validator implementations or validation rules
- Introducing self-validating value objects (see ADR-003)

## Acceptance criteria
- [ ] Email field shows an inline error when the value is invalid and the field has been edited
- [ ] Password field shows an inline error when the value is invalid and the field has been edited
- [ ] Confirm password field shows an inline error when it does not match the password field
- [ ] Profile name field shows an inline error when the value is invalid and the field has been edited
- [ ] Errors clear as soon as the field becomes valid
- [ ] Submit-time validation still runs (real-time validation is additive, not a replacement)
- [ ] ViewModel unit tests cover the real-time validation state transitions

### PR DESCRIPTION
**Pull request by timurgilfanov**
Closes #321

## Summary
Adds real-time field validation to Login and Signup screens so users see inline errors as they type, rather than only on form submission.

## Changes
- `LoginViewModel`: validates email and password on every field update; stores per-field error state and exposes it via `UiState`
- `SignupViewModel`: validates email, password, confirm password, and profile name on every field update; `updateEmail` now recalculates `passwordError` when the email field becomes valid
- Unit tests: `LoginViewModelRealTimeValidationTest` and `SignupViewModelRealTimeValidationTest` cover real-time validation state transitions

## Important Design Decision
Reuses existing `CredentialsValidator` and `ProfileNameValidator` interfaces — no validation rules duplicated in ViewModels. Real-time validation is additive: submit-time validation still runs.

## Testing
- [x] Unit tests added (`LoginViewModelRealTimeValidationTest`, `SignupViewModelRealTimeValidationTest`)
- [x] Manually tested
- [x] Edge cases considered (error clears as soon as field becomes valid, password/confirm-password cross-field dependency)

## Screenshots / Demo
N/A — logic-only change; no UI layout modifications.

## Acceptance criteria
- [x] Email field shows an inline error when the value is invalid and the field has been edited
- [x] Password field shows an inline error when the value is invalid and the field has been edited
- [x] Confirm password field shows an inline error when it does not match the password field
- [x] Profile name field shows an inline error when the value is invalid and the field has been edited
- [x] Errors clear as soon as the field becomes valid
- [x] Submit-time validation still runs (real-time validation is additive, not a replacement)
- [x] ViewModel unit tests cover the real-time validation state transitions

## Breaking Changes
None.

## Follow-ups
- [ ]

### CODE REVIEW

File: feature/auth/src/main/kotlin/timur/gilfanov/messenger/auth/ui/screen/login/LoginViewModel.kt [hunk 1]
```diff
@@ -58,23 +59,44 @@ class LoginViewModel @Inject constructor(
...
+        val emailError = (validationFailure as? CredentialsValidationError.Email)?.reason
         _state.update {
-            it.copy(email = email, emailError = null, isCredentialsValid = isCredentialsValid)
+            it.copy(
+                email = email,
+                emailError = emailError,
+                isCredentialsValid = isCredentialsValid,
+            )
         }
     }
```
[the remark this answers is not in the document]
Comment (timurgilfanov) [reply feature/auth/src/main/kotlin/timur/gilfanov/messenger/auth/ui/screen/login/LoginViewModel.kt]: False positive. `updateEmail` intentionally does **not** include `passwordError` in the `it.copy()` call, so the previous value set by `updatePassword` is preserved unchanged — which is always the correct value:

- If the user already typed an invalid password, `passwordError` stays set correctly.
- If the user hasn't touched the password field yet, `passwordError` stays `null` — preventing a premature error for an untouched field.

The proposed fix would be a regression: when email becomes invalid, `validationFailure` is `CredentialsValidationError.Email`, so the `when` branch `is CredentialsValidationError.Email, null -> null` would **clear** a visible password error simply because the user typed a bad email.
